{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.11) or chardet (3.0.4) doesn't match a supported version!\n",
      "  warnings.warn(\"urllib3 ({}) or chardet ({}) doesn't match a supported \"\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "from dotenv import load_dotenv\n",
    "from openrouteservice import client, places"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_dir = '../data/curated/'\n",
    "\n",
    "## set variables\n",
    "load_dotenv()\n",
    "tokens = []\n",
    "for i in range(1,17): \n",
    "    token = os.environ.get(f'token_{i}')\n",
    "    tokens.append(token)\n",
    "category_ids = [587, 588, 604, 607]\n",
    "buffer_size = 2000   # between 1-2000\n",
    "ptv_dict = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "## read apartment data\n",
    "property_data = json.load(open( f\"{output_dir}properties_processed.json\" ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/jwilliamsmurnane/.local/lib/python3.8/site-packages/openrouteservice/client.py:211: UserWarning: Rate limit exceeded. Retrying for the 1st time.\n",
      "  warnings.warn('Rate limit exceeded. Retrying for the {0}{1} time.'.format(retry_counter + 1,\n",
      "/home/jwilliamsmurnane/.local/lib/python3.8/site-packages/openrouteservice/client.py:211: UserWarning: Rate limit exceeded. Retrying for the 2nd time.\n",
      "  warnings.warn('Rate limit exceeded. Retrying for the {0}{1} time.'.format(retry_counter + 1,\n",
      "/home/jwilliamsmurnane/.local/lib/python3.8/site-packages/openrouteservice/client.py:211: UserWarning: Rate limit exceeded. Retrying for the 3rd time.\n",
      "  warnings.warn('Rate limit exceeded. Retrying for the {0}{1} time.'.format(retry_counter + 1,\n",
      "/home/jwilliamsmurnane/.local/lib/python3.8/site-packages/openrouteservice/client.py:211: UserWarning: Rate limit exceeded. Retrying for the 4th time.\n",
      "  warnings.warn('Rate limit exceeded. Retrying for the {0}{1} time.'.format(retry_counter + 1,\n",
      "/home/jwilliamsmurnane/.local/lib/python3.8/site-packages/openrouteservice/client.py:211: UserWarning: Rate limit exceeded. Retrying for the 5th time.\n",
      "  warnings.warn('Rate limit exceeded. Retrying for the {0}{1} time.'.format(retry_counter + 1,\n"
     ]
    },
    {
     "ename": "Timeout",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mtimeout\u001b[0m                                   Traceback (most recent call last)",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:449\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    445\u001b[0m         \u001b[39mexcept\u001b[39;00m \u001b[39mBaseException\u001b[39;00m \u001b[39mas\u001b[39;00m e:\n\u001b[1;32m    446\u001b[0m             \u001b[39m# Remove the TypeError from the exception chain in\u001b[39;00m\n\u001b[1;32m    447\u001b[0m             \u001b[39m# Python 3 (including for exceptions like SystemExit).\u001b[39;00m\n\u001b[1;32m    448\u001b[0m             \u001b[39m# Otherwise it looks like a bug in the code.\u001b[39;00m\n\u001b[0;32m--> 449\u001b[0m             six\u001b[39m.\u001b[39;49mraise_from(e, \u001b[39mNone\u001b[39;49;00m)\n\u001b[1;32m    450\u001b[0m \u001b[39mexcept\u001b[39;00m (SocketTimeout, BaseSSLError, SocketError) \u001b[39mas\u001b[39;00m e:\n",
      "File \u001b[0;32m<string>:3\u001b[0m, in \u001b[0;36mraise_from\u001b[0;34m(value, from_value)\u001b[0m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:444\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    443\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 444\u001b[0m     httplib_response \u001b[39m=\u001b[39m conn\u001b[39m.\u001b[39;49mgetresponse()\n\u001b[1;32m    445\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mBaseException\u001b[39;00m \u001b[39mas\u001b[39;00m e:\n\u001b[1;32m    446\u001b[0m     \u001b[39m# Remove the TypeError from the exception chain in\u001b[39;00m\n\u001b[1;32m    447\u001b[0m     \u001b[39m# Python 3 (including for exceptions like SystemExit).\u001b[39;00m\n\u001b[1;32m    448\u001b[0m     \u001b[39m# Otherwise it looks like a bug in the code.\u001b[39;00m\n",
      "File \u001b[0;32m/usr/lib/python3.8/http/client.py:1348\u001b[0m, in \u001b[0;36mHTTPConnection.getresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1347\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m-> 1348\u001b[0m     response\u001b[39m.\u001b[39;49mbegin()\n\u001b[1;32m   1349\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mConnectionError\u001b[39;00m:\n",
      "File \u001b[0;32m/usr/lib/python3.8/http/client.py:316\u001b[0m, in \u001b[0;36mHTTPResponse.begin\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    315\u001b[0m \u001b[39mwhile\u001b[39;00m \u001b[39mTrue\u001b[39;00m:\n\u001b[0;32m--> 316\u001b[0m     version, status, reason \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_read_status()\n\u001b[1;32m    317\u001b[0m     \u001b[39mif\u001b[39;00m status \u001b[39m!=\u001b[39m CONTINUE:\n",
      "File \u001b[0;32m/usr/lib/python3.8/http/client.py:277\u001b[0m, in \u001b[0;36mHTTPResponse._read_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    276\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_read_status\u001b[39m(\u001b[39mself\u001b[39m):\n\u001b[0;32m--> 277\u001b[0m     line \u001b[39m=\u001b[39m \u001b[39mstr\u001b[39m(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mfp\u001b[39m.\u001b[39;49mreadline(_MAXLINE \u001b[39m+\u001b[39;49m \u001b[39m1\u001b[39;49m), \u001b[39m\"\u001b[39m\u001b[39miso-8859-1\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m    278\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(line) \u001b[39m>\u001b[39m _MAXLINE:\n",
      "File \u001b[0;32m/usr/lib/python3.8/socket.py:669\u001b[0m, in \u001b[0;36mSocketIO.readinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    668\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 669\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_sock\u001b[39m.\u001b[39;49mrecv_into(b)\n\u001b[1;32m    670\u001b[0m \u001b[39mexcept\u001b[39;00m timeout:\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/contrib/pyopenssl.py:332\u001b[0m, in \u001b[0;36mWrappedSocket.recv_into\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    331\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m util\u001b[39m.\u001b[39mwait_for_read(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39msocket, \u001b[39mself\u001b[39m\u001b[39m.\u001b[39msocket\u001b[39m.\u001b[39mgettimeout()):\n\u001b[0;32m--> 332\u001b[0m     \u001b[39mraise\u001b[39;00m timeout(\u001b[39m\"\u001b[39m\u001b[39mThe read operation timed out\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m    333\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "\u001b[0;31mtimeout\u001b[0m: The read operation timed out",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mReadTimeoutError\u001b[0m                          Traceback (most recent call last)",
      "File \u001b[0;32m/usr/lib/python3/dist-packages/requests/adapters.py:439\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m chunked:\n\u001b[0;32m--> 439\u001b[0m     resp \u001b[39m=\u001b[39m conn\u001b[39m.\u001b[39;49murlopen(\n\u001b[1;32m    440\u001b[0m         method\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mmethod,\n\u001b[1;32m    441\u001b[0m         url\u001b[39m=\u001b[39;49murl,\n\u001b[1;32m    442\u001b[0m         body\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mbody,\n\u001b[1;32m    443\u001b[0m         headers\u001b[39m=\u001b[39;49mrequest\u001b[39m.\u001b[39;49mheaders,\n\u001b[1;32m    444\u001b[0m         redirect\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    445\u001b[0m         assert_same_host\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    446\u001b[0m         preload_content\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    447\u001b[0m         decode_content\u001b[39m=\u001b[39;49m\u001b[39mFalse\u001b[39;49;00m,\n\u001b[1;32m    448\u001b[0m         retries\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mmax_retries,\n\u001b[1;32m    449\u001b[0m         timeout\u001b[39m=\u001b[39;49mtimeout\n\u001b[1;32m    450\u001b[0m     )\n\u001b[1;32m    452\u001b[0m \u001b[39m# Send the request.\u001b[39;00m\n\u001b[1;32m    453\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:787\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    785\u001b[0m     e \u001b[39m=\u001b[39m ProtocolError(\u001b[39m\"\u001b[39m\u001b[39mConnection aborted.\u001b[39m\u001b[39m\"\u001b[39m, e)\n\u001b[0;32m--> 787\u001b[0m retries \u001b[39m=\u001b[39m retries\u001b[39m.\u001b[39;49mincrement(\n\u001b[1;32m    788\u001b[0m     method, url, error\u001b[39m=\u001b[39;49me, _pool\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m, _stacktrace\u001b[39m=\u001b[39;49msys\u001b[39m.\u001b[39;49mexc_info()[\u001b[39m2\u001b[39;49m]\n\u001b[1;32m    789\u001b[0m )\n\u001b[1;32m    790\u001b[0m retries\u001b[39m.\u001b[39msleep()\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/util/retry.py:550\u001b[0m, in \u001b[0;36mRetry.increment\u001b[0;34m(self, method, url, response, error, _pool, _stacktrace)\u001b[0m\n\u001b[1;32m    549\u001b[0m \u001b[39mif\u001b[39;00m read \u001b[39mis\u001b[39;00m \u001b[39mFalse\u001b[39;00m \u001b[39mor\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_is_method_retryable(method):\n\u001b[0;32m--> 550\u001b[0m     \u001b[39mraise\u001b[39;00m six\u001b[39m.\u001b[39;49mreraise(\u001b[39mtype\u001b[39;49m(error), error, _stacktrace)\n\u001b[1;32m    551\u001b[0m \u001b[39melif\u001b[39;00m read \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/packages/six.py:770\u001b[0m, in \u001b[0;36mreraise\u001b[0;34m(tp, value, tb)\u001b[0m\n\u001b[1;32m    769\u001b[0m         \u001b[39mraise\u001b[39;00m value\u001b[39m.\u001b[39mwith_traceback(tb)\n\u001b[0;32m--> 770\u001b[0m     \u001b[39mraise\u001b[39;00m value\n\u001b[1;32m    771\u001b[0m \u001b[39mfinally\u001b[39;00m:\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:703\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    702\u001b[0m \u001b[39m# Make the request on the httplib connection object.\u001b[39;00m\n\u001b[0;32m--> 703\u001b[0m httplib_response \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_make_request(\n\u001b[1;32m    704\u001b[0m     conn,\n\u001b[1;32m    705\u001b[0m     method,\n\u001b[1;32m    706\u001b[0m     url,\n\u001b[1;32m    707\u001b[0m     timeout\u001b[39m=\u001b[39;49mtimeout_obj,\n\u001b[1;32m    708\u001b[0m     body\u001b[39m=\u001b[39;49mbody,\n\u001b[1;32m    709\u001b[0m     headers\u001b[39m=\u001b[39;49mheaders,\n\u001b[1;32m    710\u001b[0m     chunked\u001b[39m=\u001b[39;49mchunked,\n\u001b[1;32m    711\u001b[0m )\n\u001b[1;32m    713\u001b[0m \u001b[39m# If we're going to release the connection in ``finally:``, then\u001b[39;00m\n\u001b[1;32m    714\u001b[0m \u001b[39m# the response doesn't need to know about the connection. Otherwise\u001b[39;00m\n\u001b[1;32m    715\u001b[0m \u001b[39m# it will also try to release it and we'll have a double-release\u001b[39;00m\n\u001b[1;32m    716\u001b[0m \u001b[39m# mess.\u001b[39;00m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:451\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    450\u001b[0m \u001b[39mexcept\u001b[39;00m (SocketTimeout, BaseSSLError, SocketError) \u001b[39mas\u001b[39;00m e:\n\u001b[0;32m--> 451\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_raise_timeout(err\u001b[39m=\u001b[39;49me, url\u001b[39m=\u001b[39;49murl, timeout_value\u001b[39m=\u001b[39;49mread_timeout)\n\u001b[1;32m    452\u001b[0m     \u001b[39mraise\u001b[39;00m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:340\u001b[0m, in \u001b[0;36mHTTPConnectionPool._raise_timeout\u001b[0;34m(self, err, url, timeout_value)\u001b[0m\n\u001b[1;32m    339\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39misinstance\u001b[39m(err, SocketTimeout):\n\u001b[0;32m--> 340\u001b[0m     \u001b[39mraise\u001b[39;00m ReadTimeoutError(\n\u001b[1;32m    341\u001b[0m         \u001b[39mself\u001b[39m, url, \u001b[39m\"\u001b[39m\u001b[39mRead timed out. (read timeout=\u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m)\u001b[39m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m timeout_value\n\u001b[1;32m    342\u001b[0m     )\n\u001b[1;32m    344\u001b[0m \u001b[39m# See the above comment about EAGAIN in Python 3. In Python 2 we have\u001b[39;00m\n\u001b[1;32m    345\u001b[0m \u001b[39m# to specifically catch it and throw the timeout error\u001b[39;00m\n",
      "\u001b[0;31mReadTimeoutError\u001b[0m: HTTPSConnectionPool(host='api.openrouteservice.org', port=443): Read timed out. (read timeout=60)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mReadTimeout\u001b[0m                               Traceback (most recent call last)",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/openrouteservice/client.py:186\u001b[0m, in \u001b[0;36mClient.request\u001b[0;34m(self, url, get_params, first_request_time, retry_counter, requests_kwargs, post_json, dry_run)\u001b[0m\n\u001b[1;32m    185\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 186\u001b[0m     response \u001b[39m=\u001b[39m requests_method(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_base_url \u001b[39m+\u001b[39;49m authed_url,\n\u001b[1;32m    187\u001b[0m                                \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mfinal_requests_kwargs)\n\u001b[1;32m    188\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_req \u001b[39m=\u001b[39m response\u001b[39m.\u001b[39mrequest\n",
      "File \u001b[0;32m/usr/lib/python3/dist-packages/requests/sessions.py:581\u001b[0m, in \u001b[0;36mSession.post\u001b[0;34m(self, url, data, json, **kwargs)\u001b[0m\n\u001b[1;32m    571\u001b[0m \u001b[39mr\u001b[39m\u001b[39m\"\"\"Sends a POST request. Returns :class:`Response` object.\u001b[39;00m\n\u001b[1;32m    572\u001b[0m \n\u001b[1;32m    573\u001b[0m \u001b[39m:param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    578\u001b[0m \u001b[39m:rtype: requests.Response\u001b[39;00m\n\u001b[1;32m    579\u001b[0m \u001b[39m\"\"\"\u001b[39;00m\n\u001b[0;32m--> 581\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mrequest(\u001b[39m'\u001b[39;49m\u001b[39mPOST\u001b[39;49m\u001b[39m'\u001b[39;49m, url, data\u001b[39m=\u001b[39;49mdata, json\u001b[39m=\u001b[39;49mjson, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n",
      "File \u001b[0;32m/usr/lib/python3/dist-packages/requests/sessions.py:533\u001b[0m, in \u001b[0;36mSession.request\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    532\u001b[0m send_kwargs\u001b[39m.\u001b[39mupdate(settings)\n\u001b[0;32m--> 533\u001b[0m resp \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49msend(prep, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49msend_kwargs)\n\u001b[1;32m    535\u001b[0m \u001b[39mreturn\u001b[39;00m resp\n",
      "File \u001b[0;32m/usr/lib/python3/dist-packages/requests/sessions.py:646\u001b[0m, in \u001b[0;36mSession.send\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    645\u001b[0m \u001b[39m# Send the request\u001b[39;00m\n\u001b[0;32m--> 646\u001b[0m r \u001b[39m=\u001b[39m adapter\u001b[39m.\u001b[39;49msend(request, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n\u001b[1;32m    648\u001b[0m \u001b[39m# Total elapsed time of the request (approximately)\u001b[39;00m\n",
      "File \u001b[0;32m/usr/lib/python3/dist-packages/requests/adapters.py:529\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    528\u001b[0m \u001b[39melif\u001b[39;00m \u001b[39misinstance\u001b[39m(e, ReadTimeoutError):\n\u001b[0;32m--> 529\u001b[0m     \u001b[39mraise\u001b[39;00m ReadTimeout(e, request\u001b[39m=\u001b[39mrequest)\n\u001b[1;32m    530\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "\u001b[0;31mReadTimeout\u001b[0m: HTTPSConnectionPool(host='api.openrouteservice.org', port=443): Read timed out. (read timeout=60)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mTimeout\u001b[0m                                   Traceback (most recent call last)",
      "\u001b[1;32m/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb Cell 4\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=12'>13</a>\u001b[0m ors \u001b[39m=\u001b[39m client\u001b[39m.\u001b[39mClient(key\u001b[39m=\u001b[39mtoken)\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=13'>14</a>\u001b[0m query \u001b[39m=\u001b[39m {\u001b[39m'\u001b[39m\u001b[39mrequest\u001b[39m\u001b[39m'\u001b[39m: \u001b[39m'\u001b[39m\u001b[39mpois\u001b[39m\u001b[39m'\u001b[39m,\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=14'>15</a>\u001b[0m         \u001b[39m'\u001b[39m\u001b[39mgeojson\u001b[39m\u001b[39m'\u001b[39m: {\u001b[39m'\u001b[39m\u001b[39mtype\u001b[39m\u001b[39m'\u001b[39m:\u001b[39m'\u001b[39m\u001b[39mPoint\u001b[39m\u001b[39m'\u001b[39m,\u001b[39m'\u001b[39m\u001b[39mcoordinates\u001b[39m\u001b[39m'\u001b[39m:coords},\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=15'>16</a>\u001b[0m         \u001b[39m'\u001b[39m\u001b[39mbuffer\u001b[39m\u001b[39m'\u001b[39m: buffer_size,\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=16'>17</a>\u001b[0m         \u001b[39m'\u001b[39m\u001b[39mfilter_category_ids\u001b[39m\u001b[39m'\u001b[39m: category_ids,\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=17'>18</a>\u001b[0m         \u001b[39m'\u001b[39m\u001b[39msortby\u001b[39m\u001b[39m'\u001b[39m:\u001b[39m'\u001b[39m\u001b[39mdistance\u001b[39m\u001b[39m'\u001b[39m,\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=18'>19</a>\u001b[0m         \u001b[39m'\u001b[39m\u001b[39mlimit\u001b[39m\u001b[39m'\u001b[39m: \u001b[39m4\u001b[39m}\n\u001b[0;32m---> <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=19'>20</a>\u001b[0m features \u001b[39m=\u001b[39m ors\u001b[39m.\u001b[39;49mplaces(\u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mquery)[\u001b[39m'\u001b[39m\u001b[39mfeatures\u001b[39m\u001b[39m'\u001b[39m]\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=21'>22</a>\u001b[0m \u001b[39m## distances - direct meters\u001b[39;00m\n\u001b[1;32m     <a href='vscode-notebook-cell://wsl%2Bubuntu/mnt/c/Users/jwill/Desktop/UNI/2022_SEM2/ads/generic-real-estate-consulting-project-group-42/notebooks/download_ptv.ipynb#W3sdnNjb2RlLXJlbW90ZQ%3D%3D?line=22'>23</a>\u001b[0m distances \u001b[39m=\u001b[39m []\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/openrouteservice/client.py:299\u001b[0m, in \u001b[0;36m_make_api_method.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    296\u001b[0m \u001b[39m@functools\u001b[39m\u001b[39m.\u001b[39mwraps(func)\n\u001b[1;32m    297\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mwrapper\u001b[39m(\u001b[39m*\u001b[39margs, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs):\n\u001b[1;32m    298\u001b[0m     args[\u001b[39m0\u001b[39m]\u001b[39m.\u001b[39m_extra_params \u001b[39m=\u001b[39m kwargs\u001b[39m.\u001b[39mpop(\u001b[39m\"\u001b[39m\u001b[39mextra_params\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39mNone\u001b[39;00m)\n\u001b[0;32m--> 299\u001b[0m     result \u001b[39m=\u001b[39m func(\u001b[39m*\u001b[39;49margs, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49mkwargs)\n\u001b[1;32m    300\u001b[0m     \u001b[39mtry\u001b[39;00m:\n\u001b[1;32m    301\u001b[0m         \u001b[39mdel\u001b[39;00m args[\u001b[39m0\u001b[39m]\u001b[39m.\u001b[39m_extra_params\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/openrouteservice/places.py:115\u001b[0m, in \u001b[0;36mplaces\u001b[0;34m(client, request, geojson, bbox, buffer, filter_category_ids, filter_category_group_ids, filters_custom, limit, sortby, validate, dry_run)\u001b[0m\n\u001b[1;32m    112\u001b[0m     \u001b[39mif\u001b[39;00m sortby:\n\u001b[1;32m    113\u001b[0m         params[\u001b[39m'\u001b[39m\u001b[39msortby\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m=\u001b[39m sortby\n\u001b[0;32m--> 115\u001b[0m \u001b[39mreturn\u001b[39;00m client\u001b[39m.\u001b[39;49mrequest(\u001b[39m'\u001b[39;49m\u001b[39m/pois\u001b[39;49m\u001b[39m'\u001b[39;49m, {}, post_json\u001b[39m=\u001b[39;49mparams, dry_run\u001b[39m=\u001b[39;49mdry_run)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/openrouteservice/client.py:216\u001b[0m, in \u001b[0;36mClient.request\u001b[0;34m(self, url, get_params, first_request_time, retry_counter, requests_kwargs, post_json, dry_run)\u001b[0m\n\u001b[1;32m    211\u001b[0m warnings\u001b[39m.\u001b[39mwarn(\u001b[39m'\u001b[39m\u001b[39mRate limit exceeded. Retrying for the \u001b[39m\u001b[39m{0}\u001b[39;00m\u001b[39m{1}\u001b[39;00m\u001b[39m time.\u001b[39m\u001b[39m'\u001b[39m\u001b[39m.\u001b[39mformat(retry_counter \u001b[39m+\u001b[39m \u001b[39m1\u001b[39m,\n\u001b[1;32m    212\u001b[0m                                                                           get_ordinal(retry_counter \u001b[39m+\u001b[39m \u001b[39m1\u001b[39m)),\n\u001b[1;32m    213\u001b[0m               \u001b[39mUserWarning\u001b[39;00m,\n\u001b[1;32m    214\u001b[0m               stacklevel\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[1;32m    215\u001b[0m \u001b[39m# Retry request.\u001b[39;00m\n\u001b[0;32m--> 216\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mrequest(url, get_params, first_request_time,\n\u001b[1;32m    217\u001b[0m                     retry_counter \u001b[39m+\u001b[39;49m \u001b[39m1\u001b[39;49m, requests_kwargs,\n\u001b[1;32m    218\u001b[0m                     post_json)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/openrouteservice/client.py:216\u001b[0m, in \u001b[0;36mClient.request\u001b[0;34m(self, url, get_params, first_request_time, retry_counter, requests_kwargs, post_json, dry_run)\u001b[0m\n\u001b[1;32m    211\u001b[0m warnings\u001b[39m.\u001b[39mwarn(\u001b[39m'\u001b[39m\u001b[39mRate limit exceeded. Retrying for the \u001b[39m\u001b[39m{0}\u001b[39;00m\u001b[39m{1}\u001b[39;00m\u001b[39m time.\u001b[39m\u001b[39m'\u001b[39m\u001b[39m.\u001b[39mformat(retry_counter \u001b[39m+\u001b[39m \u001b[39m1\u001b[39m,\n\u001b[1;32m    212\u001b[0m                                                                           get_ordinal(retry_counter \u001b[39m+\u001b[39m \u001b[39m1\u001b[39m)),\n\u001b[1;32m    213\u001b[0m               \u001b[39mUserWarning\u001b[39;00m,\n\u001b[1;32m    214\u001b[0m               stacklevel\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[1;32m    215\u001b[0m \u001b[39m# Retry request.\u001b[39;00m\n\u001b[0;32m--> 216\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mrequest(url, get_params, first_request_time,\n\u001b[1;32m    217\u001b[0m                     retry_counter \u001b[39m+\u001b[39;49m \u001b[39m1\u001b[39;49m, requests_kwargs,\n\u001b[1;32m    218\u001b[0m                     post_json)\n",
      "    \u001b[0;31m[... skipping similar frames: Client.request at line 216 (2 times)]\u001b[0m\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/openrouteservice/client.py:216\u001b[0m, in \u001b[0;36mClient.request\u001b[0;34m(self, url, get_params, first_request_time, retry_counter, requests_kwargs, post_json, dry_run)\u001b[0m\n\u001b[1;32m    211\u001b[0m warnings\u001b[39m.\u001b[39mwarn(\u001b[39m'\u001b[39m\u001b[39mRate limit exceeded. Retrying for the \u001b[39m\u001b[39m{0}\u001b[39;00m\u001b[39m{1}\u001b[39;00m\u001b[39m time.\u001b[39m\u001b[39m'\u001b[39m\u001b[39m.\u001b[39mformat(retry_counter \u001b[39m+\u001b[39m \u001b[39m1\u001b[39m,\n\u001b[1;32m    212\u001b[0m                                                                           get_ordinal(retry_counter \u001b[39m+\u001b[39m \u001b[39m1\u001b[39m)),\n\u001b[1;32m    213\u001b[0m               \u001b[39mUserWarning\u001b[39;00m,\n\u001b[1;32m    214\u001b[0m               stacklevel\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[1;32m    215\u001b[0m \u001b[39m# Retry request.\u001b[39;00m\n\u001b[0;32m--> 216\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mrequest(url, get_params, first_request_time,\n\u001b[1;32m    217\u001b[0m                     retry_counter \u001b[39m+\u001b[39;49m \u001b[39m1\u001b[39;49m, requests_kwargs,\n\u001b[1;32m    218\u001b[0m                     post_json)\n",
      "File \u001b[0;32m~/.local/lib/python3.8/site-packages/openrouteservice/client.py:191\u001b[0m, in \u001b[0;36mClient.request\u001b[0;34m(self, url, get_params, first_request_time, retry_counter, requests_kwargs, post_json, dry_run)\u001b[0m\n\u001b[1;32m    188\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_req \u001b[39m=\u001b[39m response\u001b[39m.\u001b[39mrequest\n\u001b[1;32m    190\u001b[0m \u001b[39mexcept\u001b[39;00m requests\u001b[39m.\u001b[39mexceptions\u001b[39m.\u001b[39mTimeout:\n\u001b[0;32m--> 191\u001b[0m     \u001b[39mraise\u001b[39;00m exceptions\u001b[39m.\u001b[39mTimeout()\n\u001b[1;32m    193\u001b[0m \u001b[39mif\u001b[39;00m response\u001b[39m.\u001b[39mstatus_code \u001b[39min\u001b[39;00m _RETRIABLE_STATUSES:\n\u001b[1;32m    194\u001b[0m     \u001b[39m# Retry request.\u001b[39;00m\n\u001b[1;32m    195\u001b[0m     warnings\u001b[39m.\u001b[39mwarn(\u001b[39m'\u001b[39m\u001b[39mServer down.\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39mRetrying for the \u001b[39m\u001b[39m{0}\u001b[39;00m\u001b[39m{1}\u001b[39;00m\u001b[39m time.\u001b[39m\u001b[39m'\u001b[39m\u001b[39m.\u001b[39mformat(retry_counter \u001b[39m+\u001b[39m \u001b[39m1\u001b[39m,\n\u001b[1;32m    196\u001b[0m                                                                        get_ordinal(retry_counter \u001b[39m+\u001b[39m \u001b[39m1\u001b[39m)),\n\u001b[1;32m    197\u001b[0m                   \u001b[39mUserWarning\u001b[39;00m,\n\u001b[1;32m    198\u001b[0m                   stacklevel\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n",
      "\u001b[0;31mTimeout\u001b[0m: "
     ]
    }
   ],
   "source": [
    "for token in tokens:\n",
    "    ## set x and y\n",
    "    x = (tokens.index(token)*50)\n",
    "    y = (tokens.index(token)*50)+50\n",
    "\n",
    "    for index in range(x,y):\n",
    "        if index > len(property_data['Coordinates'].keys()):\n",
    "            break\n",
    "        backwards = list(map(float,property_data['Coordinates'][str(index)][1:-1].split(',')))\n",
    "        coords = [backwards[1], backwards[0]]\n",
    "\n",
    "        ## query code\n",
    "        ors = client.Client(key=token)\n",
    "        query = {'request': 'pois',\n",
    "                'geojson': {'type':'Point','coordinates':coords},\n",
    "                'buffer': buffer_size,\n",
    "                'filter_category_ids': category_ids,\n",
    "                'sortby':'distance',\n",
    "                'limit': 4}\n",
    "        features = ors.places(**query)['features']\n",
    "\n",
    "        ## distances - direct meters\n",
    "        distances = []\n",
    "        for poi in features:\n",
    "            distances.append(poi['properties']['distance'])\n",
    "        \n",
    "        ## add information\n",
    "        ptv_dict[str(index)] = distances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'0': [41.06558888, 94.70841003, 131.18498313],\n",
       " '1': [144.95323909, 157.01732523, 166.74847649],\n",
       " '2': [26.20177793, 32.82987166, 54.08487916],\n",
       " '3': [17.50050173, 57.57980144, 96.76005852],\n",
       " '4': [48.06643745, 70.60349219, 79.34162645],\n",
       " '5': [81.83820034, 99.02751071, 122.65489126],\n",
       " '6': [55.35311455, 55.56729387, 55.90871754],\n",
       " '7': [61.03556356, 118.89520895, 147.90045854],\n",
       " '8': [72.60070446, 75.91353511, 115.09935415],\n",
       " '9': [57.30905232, 65.01360221, 104.66903829],\n",
       " '10': [175.37710899, 193.89040545, 199.02513883],\n",
       " '11': [48.06643745, 70.60349219, 79.34162645],\n",
       " '12': [48.06643745, 70.60349219, 79.34162645],\n",
       " '13': [59.69829384, 100.25937212, 128.58180338],\n",
       " '14': [83.36413507, 152.54942177, 154.17434406],\n",
       " '15': [31.66243137, 40.06651561, 49.46019789],\n",
       " '16': [64.93078674, 66.8099241, 67.3655901],\n",
       " '17': [110.48181552, 115.82052191, 134.3972878],\n",
       " '18': [103.04893255, 106.77079889, 113.53842745],\n",
       " '19': [55.35311455, 55.56729387, 55.90871754],\n",
       " '20': [70.7428171, 74.73475915, 95.23372607],\n",
       " '21': [64.93078674, 66.8099241, 67.3655901],\n",
       " '22': [87.29258374, 96.66553016, 185.80606188],\n",
       " '23': [87.29258374, 96.66553016, 185.80606188],\n",
       " '24': [34.97531792, 38.52499445, 65.59339678],\n",
       " '25': [45.92559958, 58.72432434, 88.60971668],\n",
       " '26': [100.53943279, 126.68213037, 169.85560278],\n",
       " '27': [122.62806957, 155.15664002, 163.73293574],\n",
       " '28': [42.27855017, 57.18674161, 84.95827972],\n",
       " '29': [70.0629339, 73.60104927, 88.96887846],\n",
       " '30': [32.08694514, 38.33042372, 114.31306852],\n",
       " '31': [255.87520186, 257.71389596, 261.94880214],\n",
       " '32': [26.69569932, 26.69569936, 30.43160976],\n",
       " '33': [72.74540071, 121.34140365, 169.58041835],\n",
       " '34': [48.06643745, 70.60349219, 79.34162645],\n",
       " '35': [30.26209862, 52.80235153, 76.85532002],\n",
       " '36': [36.12624276, 45.27273971, 82.35916851],\n",
       " '37': [168.52951633, 174.31239069, 182.43791756],\n",
       " '38': [72.4711598, 85.36331318, 90.02944804],\n",
       " '39': [115.59494006, 124.08342647, 127.6806321],\n",
       " '40': [73.93780047, 77.15036972, 88.86472132],\n",
       " '41': [164.59165456, 167.42096015, 239.05613104],\n",
       " '42': [120.9116544, 130.75148964, 247.84807426],\n",
       " '43': [319.53101102, 360.50103013, 366.38460164],\n",
       " '44': [88.80292335, 98.31582969, 193.47429668],\n",
       " '45': [185.78141818, 187.89426246, 212.01068491],\n",
       " '46': [65.16644443, 115.87580507, 117.75808429],\n",
       " '47': [271.25452708, 277.64791779, 343.60963657],\n",
       " '48': [102.44574181, 104.06795828, 143.16680171],\n",
       " '49': [145.60795538, 151.75093901, 246.9043595],\n",
       " '50': [80.85288006, 159.79447254, 162.37441899],\n",
       " '51': [100.84212476, 105.17537667, 113.78075023],\n",
       " '52': [86.54965591, 96.22984368, 231.32845569],\n",
       " '53': [68.97418937, 75.3526251, 145.16690868],\n",
       " '54': [319.53101102, 360.50103013, 366.38460164],\n",
       " '55': [195.02743141, 199.29649103, 200.22536941],\n",
       " '56': [205.52191615, 225.65633994, 228.34459911],\n",
       " '57': [227.88881237, 237.80160199, 315.47109182],\n",
       " '58': [106.46853212, 119.56286327, 139.4593601],\n",
       " '59': [65.16644443, 115.87580507, 117.75808429],\n",
       " '60': [88.80292335, 98.31582969, 193.47429668],\n",
       " '61': [48.01616133, 56.79033513, 172.32037875],\n",
       " '62': [350.13197605, 360.03614808, 425.51403081],\n",
       " '63': [146.11743028, 149.23496861, 156.62689611],\n",
       " '64': [170.70575222, 178.21901026, 212.24664481],\n",
       " '65': [95.62592223, 167.17989793, 205.02603831],\n",
       " '66': [195.02743141, 199.29649103, 200.22536941],\n",
       " '67': [131.44949584, 136.48098637, 191.41743606],\n",
       " '68': [62.81087299, 96.58583942, 103.83812163],\n",
       " '69': [142.07254924, 151.85925248, 271.47430062],\n",
       " '70': [154.47615558, 186.58257747, 196.58231785],\n",
       " '71': [271.25452708, 277.64791779, 343.60963657],\n",
       " '72': [98.62311183, 133.24968357, 177.80472776],\n",
       " '73': [119.51778825, 152.79711721, 163.46014937],\n",
       " '74': [219.22838915, 222.05353769, 228.75197251],\n",
       " '75': [129.28902017, 156.18015506, 186.34572349],\n",
       " '76': [106.46853212, 119.56286327, 139.4593601],\n",
       " '77': [73.22478859, 107.81253829, 297.28737014],\n",
       " '78': [48.01616133, 56.79033513, 172.32037875],\n",
       " '79': [206.20109106, 214.17420022, 256.88407816],\n",
       " '80': [88.80292335, 98.31582969, 193.47429668],\n",
       " '81': [65.26106865, 82.40142206, 182.04322917],\n",
       " '82': [65.26106865, 82.40142206, 182.04322917],\n",
       " '83': [65.26106865, 82.40142206, 182.04322917],\n",
       " '84': [146.77315694, 154.69784452, 157.57357148],\n",
       " '85': [65.26106865, 82.40142206, 182.04322917],\n",
       " '86': [51.04231706, 111.58849575, 174.60543878],\n",
       " '87': [27.24098248, 29.66422275, 49.35563778],\n",
       " '88': [65.26106865, 82.40142206, 182.04322917],\n",
       " '89': [167.22381365, 184.4638379, 210.42300172],\n",
       " '90': [54.04017997, 125.85450322, 131.02195777],\n",
       " '91': [176.79803701, 198.86702414, 202.77830335],\n",
       " '92': [168.40171483, 175.1109211, 232.99320811],\n",
       " '93': [343.0611495, 343.26401298, 360.86507859],\n",
       " '94': [40.50498778, 147.96246831, 160.11036205],\n",
       " '95': [210.49864584, 225.30608741, 227.10705891],\n",
       " '96': [192.76545502, 211.50148043, 215.10621127],\n",
       " '97': [65.26106865, 82.40142206, 182.04322917],\n",
       " '98': [139.41367231, 174.21047137, 245.79611207],\n",
       " '99': [100.84550271, 129.65713003, 141.96547711],\n",
       " '100': [96.07004242, 107.35305874, 120.75678525],\n",
       " '101': [100.9100489, 111.96787058, 146.74103454],\n",
       " '102': [100.9100489, 111.96787058, 146.74103454],\n",
       " '103': [136.15043003, 148.75516917, 163.83928515],\n",
       " '104': [167.22381365, 184.4638379, 210.42300172],\n",
       " '105': [65.26106865, 82.40142206, 182.04322917],\n",
       " '106': [68.19444672, 93.11969841, 146.2265791],\n",
       " '107': [68.89353935, 94.94274955, 115.98937724],\n",
       " '108': [68.19444672, 93.11969841, 146.2265791],\n",
       " '109': [27.24098248, 29.66422275, 49.35563778],\n",
       " '110': [100.9100489, 111.96787058, 146.74103454],\n",
       " '111': [114.58000708, 157.44435549, 183.35172431],\n",
       " '112': [68.19444672, 93.11969841, 146.2265791],\n",
       " '113': [100.9100489, 111.96787058, 146.74103454],\n",
       " '114': [139.41367231, 174.21047137, 245.79611207],\n",
       " '115': [84.19640457, 125.14789832, 206.52461853],\n",
       " '116': [68.19444672, 93.11969841, 146.2265791],\n",
       " '117': [96.07004242, 107.35305874, 120.75678525],\n",
       " '118': [56.1274688, 87.21469815, 176.88519827],\n",
       " '119': [92.86863074, 136.33654286, 147.83454095],\n",
       " '120': [211.8047521, 220.42571968, 224.18116566],\n",
       " '121': [65.26106865, 82.40142206, 182.04322917],\n",
       " '122': [82.4615844, 122.76175121, 171.02871359],\n",
       " '123': [159.07228091, 168.38139497, 344.94923173],\n",
       " '124': [144.18329666, 155.5623163, 158.58922417],\n",
       " '125': [139.69285485, 156.79230409, 167.81640244],\n",
       " '126': [118.7892342, 126.97176126, 146.81901237],\n",
       " '127': [154.15037037, 157.70762709, 299.25988283],\n",
       " '128': [79.38865091, 94.9540061, 100.88956224],\n",
       " '129': [128.69413317, 132.46017889, 269.00595462]}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ptv_dict"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8.10 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "916dbcbb3f70747c44a77c7bcd40155683ae19c65e1c03b4aa3499c5328201f1"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
